contracts/usul/BaseTokenVoting.sol

```diff
@@ -0,0 +1,172 @@
+    function _updateTimelockPeriod(uint256 _newTimelockPeriod) internal {
+        timelockPeriod = _newTimelockPeriod;
+
+        emit TimelockPeriodUpdated(_newTimelockPeriod);
+    }
+
+    /// @notice Returns true if an account has voted on the specified proposal
+    /// @param _proposalId The ID of the proposal to check
+    /// @param _account The account address to check
+    /// @return bool Returns true if the account has already voted on the proposal
+    function hasVoted(
+        uint256 _proposalId,
+        address _account
+    ) public view returns (bool) {
+        return proposals[_proposalId].hasVoted[_account];
+    }
+
+    /// @notice Function for counting a vote for a proposal, can only be called internally
+    /// @param _proposalId The ID of the proposal
+    /// @param _voter The address of the account casting the vote
+    /// @param _support Indicates vote support, which can be "No", "Yes", or "Abstain"
+    /// @param _weight The amount of voting weight cast
+    function _vote(
+        uint256 _proposalId,
+        address _voter,
+        uint8 _support,
+        uint256 _weight
+    ) internal {
+        require(
+            proposals[_proposalId].deadline != 0,
+            "Proposal has not been submitted yet"
+        );
+        require(
+            block.timestamp <= proposals[_proposalId].deadline,
+            "Voting period has passed"
+        );
+        require(
+            !proposals[_proposalId].hasVoted[_voter],
+            "Voter has already voted"
+        );
+
+        proposals[_proposalId].hasVoted[_voter] = true;
+
+        if (_support == uint8(VoteType.NO)) {
+            proposals[_proposalId].noVotes += _weight;
+        } else if (_support == uint8(VoteType.YES)) {
+            proposals[_proposalId].yesVotes += _weight;
+        } else if (_support == uint8(VoteType.ABSTAIN)) {
+            proposals[_proposalId].abstainVotes += _weight;
+        } else {
+            revert("Invalid value for enum VoteType");
+        }
+
+        emit Voted(_voter, _proposalId, _support, _weight);
+    }
+
+    /// @notice Called by the proposal module, this notifes the strategy of a new proposal
+    /// @param _data Any extra data to pass to the voting strategy
+    function initializeProposal(
+        bytes memory _data
+    ) external virtual override onlyUsul {
+        uint256 proposalId = abi.decode(_data, (uint256));
+
+        proposals[proposalId].deadline = votingPeriod + block.timestamp;
+        proposals[proposalId].startBlock = block.number;
+
+        emit ProposalReceived(proposalId, block.timestamp);
+    }
+
+    /// @notice Calls the Usul module to notify that a quorum has been reached
+    /// @notice Timelocks the proposal and starts timelock period
+    /// @param _proposalId The ID of the proposal to timelock
+    function timelockProposal(uint256 _proposalId) public virtual override {
+        require(isPassed(_proposalId));
```

Comment:
I saw this missing message in a PR I did for some formatting tweaks to this:

https://github.com/decent-dao/fractal-contracts/pull/32

But I can't get the tests to run locally so I can fix all the ones that broke because I tweaked all the messages..

Might need your help @tbwebb22 on Monday to get set up locally to run the tests

Reply:
Yeah so I'm guessing the tests aren't running due to not having the provider set in .env.

I did push up a change that adds a comment to this require statement